Known bugs
===========

Path
----

Sometimes, the path is not correct (startdir). Not sure to know the reason.
Maybe it comes from the curdir when the script is run.


Similar links in the same file
------------------------------

If (at least) one file contains several times the same link, each link receive several archive tags.
This is due to the parallel threading I guess


Links in parenthesis
--------------------

if the line contains:
 (http://toto.org), it fails.

* Reproduced in test_processtext
 -> a url in () gets an incorrect results.
 -> Moreover, libzimsciunto contains utils.protect
 --> I think it should protect ( and )
 --> It must be in ZimArchivist/ instead...


Traceback

Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/lib/python3.3/threading.py", line 639, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.3/site-packages/libzimsciunto/zimnotes.py", line 105, in run
    (error, new_text) = self.process_text_method(original_text, *self.args)
  File "/usr/lib/python3.3/site-packages/ZimArchivist/processtext.py", line 58, in process_text
    if not editline.link_archive_status(url, copy_text):
  File "/usr/lib/python3.3/site-packages/ZimArchivist/editline.py", line 47, in link_archive_status
    link_archived = re.compile(utils.protect(url) + '(\|.*\]\])?\s\[\[.*\|\(Archive\)\]\]')
  File "/usr/lib/python3.3/re.py", line 214, in compile
    return _compile(pattern, flags)
  File "/usr/lib/python3.3/functools.py", line 258, in wrapper
    result = user_function(*args, **kwds)
  File "/usr/lib/python3.3/re.py", line 274, in _compile
    return sre_compile.compile(pattern, flags)
  File "/usr/lib/python3.3/sre_compile.py", line 493, in compile
    p = sre_parse.parse(p, flags)
  File "/usr/lib/python3.3/sre_parse.py", line 729, in parse
    raise error("unbalanced parenthesis")
sre_constants.error: unbalanced parenthesis



